src/ray/gcs/redis_context.h

```diff
@@ -129,7 +138,7 @@ class RedisContext {
  public:
   RedisContext(instrumented_io_context &io_service);
 
-  ~RedisContext();
+  virtual ~RedisContext();
```

Comment:
why virtual?

Reply:
I'll remove this and the other virtual and just put them in the stacked change that actually uses them, since that's your point.

However, the real reason (you probably know this) that this one is virtual is because when a class member function is virtual, there is a risk of resource leak if the destructor is not virtual -- if you destroy an object initialized with `Parent a = new Child()` a nonvirtual destructor will resolve to `~Parent`, so if Child has additional resources that are cleaned up in `~Child`, they get leaked. So this is virtual because RunArgvAsync is virtual.

It doesn't matter in this specific case, but we've turned on an error-on-warning for it.